api-registration/server/index.js

```diff
@@ -0,0 +1,64 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+  const sql = `
+        insert into "users" ("username", "hashedPassword")
+        values ($1, $2)
+        returning *
+      `;
+
+  argon2.hash(password)
+    .then(hashedPasswords => {
+      const params = [username, hashedPasswords];
+      db.query(sql, params)
+        .then(result => {
+          const [userInfo] = result.rows;
+          res.status(201).json(userInfo);
+        })
```

Comment:
The client should not receive the hashed password. You can specify which columns to return in the `returning` clause of your sql query.

Reply:
Ah, will revise the code at once!